### PR TITLE
feat(nx): update usage of facades and barrel exports with NgRx

### PR DESCRIPTION
**Pull request by brandonroberts**
_[Please make sure you have read the submission guidelines before posting an PR](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#submit-pr)_

> _Please make sure that your commit message follows our format._

> Example: `fix(nx): must begin with lowercase`

### CODE REVIEW

File: packages/angular/src/schematics/ngrx/creator-files/__directory__/__fileName__.facade.ts__tmpl__ [hunk 1]
```diff
@@ -15,7 +14,7 @@ export class <%= className %>Facade {
 
   constructor(private store: Store<from<%= className %>.<%= className %>PartialState>) { }
 
-  loadAll() {
-    this.store.dispatch(<%= className %>Actions.load<%= className %>());
+  dispatch(action: Action) {
+    this.store.dispatch(action);
```
Comment (vsavkin) [inline packages/angular/src/schematics/ngrx/creator-files/__directory__/__fileName__.facade.ts__tmpl__]: Can you expand on this change? My understanding is the benefits of using the facade is that it give you a concrete API of dispatching actions and selecting things. If we simply pass actions through, why having a facade?

Comment (brandonroberts) [reply packages/angular/src/schematics/ngrx/creator-files/__directory__/__fileName__.facade.ts__tmpl__]: We still want to encourage good NgRx practices for those who want to use facades.

In general, we recommend:

- Actions should be unique in the source/event they are describing.
- Actions have a unique dispatch point, and are not reused.
- Actions should live close to the place they are dispatched from.

The update to the facade encourages those recommendations.

Comment (vsavkin) [reply packages/angular/src/schematics/ngrx/creator-files/__directory__/__fileName__.facade.ts__tmpl__]: I agree with recommendations. You are saying that the current impl of facades violates "Actions should live close to the place they are dispatched from", right? 

My concern is that as far as I understand facades are usually mocked up in tests like this (I think this is one of the main reasons why folks use facades):

```
it('should create an item', () => {
   const facade = jasmine.createObj();
   const c = new MyComponent(facade);
   c.someClickHander();
   expect(facade.loadBlah).toHaveBeenCalledWith('somevalue');
});
```

with this change you will have to do:

```
it('should create an item', () => {
   const facade = jasmine.createObj();
   const c = new MyComponent(facade);
   c.someClickHander();
   expect(c.dispatch.calls.mostRecent().args[0].valueField).toEqual('somevalue');
});
```
I imagine people who care about facades may not be OK with it.

What do you think?

**Comment by brandonroberts** [comment]
I don't imagine the test being that verbose to get the dispatched value.

```
it('should dispatch an action to create an item', () => {
   const facade = jasmine.createObj();
   const c = new MyComponent(facade);
   const myAction = someActionCreator({ id: 1 });
   c.someClickHander();
   expect(facade.dispatch).toHaveBeenCalledWith(some action);
});
```

**Pull request merged**